internal/generator/vector/output/loki/loki.go

```diff
@@ -233,3 +234,24 @@ func BasicAuth(o logging.OutputSpec, secret *corev1.Secret) []Element {
 
 	return conf
 }
+
+func BearerTokenAuth(o logging.OutputSpec, secret *corev1.Secret) []Element {
+	conf := []Element{}
+	if o.Secret == nil {
+		// internal loki, use bearer token of logcollector service account
+		// the secret contains the token itself
+		if secret != nil && security.HasBearerTokenFileKey(secret) {
+			conf = append(conf, BasicAuthConf{
+				Desc:        "Bearer Auth Config",
+				ComponentID: strings.ToLower(vectorhelpers.Replacer.Replace(o.Name)),
+			})
+			bt := BearerToken{
+				Token: security.GetFromSecret(secret, constants.BearerTokenFileKey),
```

Comment:
@vimalk78 I still don't understand why we are storing a file name in the secret rather than mounting the secret data.
If this is in order to use tokens already mounted (e.g. SA token) then we should change the name from 'token', but I don't think that the user should have to know the path where this gets mounted. E.g. if we are picking up an SA token then maybe it should be 'tokenServiceAccount' with the SA name, and the CLO can figure out where that is mounted.

Reply:
just the name of the variable is `BearerTokenFileKey`, the value is just `"token"` . Actually this step is just reading the key from the secret.